task_processing/interfaces/runner.py

```diff
@@ -14,9 +16,12 @@ class Runner(object):
     """
 
     @abc.abstractmethod
-    def run(self, task_config):
+    def run(self, task_config, task_id=None):
         pass
 
     @abc.abstractmethod
     def kill(self, task_id):
         pass
+
+    def new_task_id(self):
```

Comment:
task_id is something we don't want users to touch at all. but if they do - let's give them a function that makes it easy.